test/datalog_tests/server_api/tests/deltas.rs

```diff
@@ -1,58 +1,113 @@
+use std::any::Any;
 use std::sync::Mutex;
+use std::time::Duration;
 
+use differential_datalog::program::Update;
 use differential_datalog::record::{Record, RelIdentifier, UpdCmd};
 use observe::Observable;
 use observe::Observer;
+use observe::SharedObserver;
+use tcp_channel::TcpReceiver;
+use tcp_channel::TcpSender;
 
 use server_api_ddlog::api::*;
-use server_api_ddlog::server;
+use server_api_ddlog::server::*;
 use server_api_ddlog::Relations::*;
+use server_api_ddlog::Value;
 
 use maplit::hashmap;
 use maplit::hashset;
 
+use waitfor::wait_for;
+
+macro_rules! DeltaTest {
+    ( $($setup_fn:tt)* ) => {{
+        // We need a global here (and cannot, say, close over some local
+        // from a lambda expression) because our callback has to implement
+        // Clone and that is not possible for a closure capturing its
+        // environment in a mutable way.
+        // We need an option because Vec::new is not const and so we can't
+        // use it to initialize a static.
+        static mut DELTAS: Option<Mutex<Vec<(usize, Record)>>> = None;
```

Comment:
would `lazy_static` help here?

Reply:
I tried it, but the compiler threw some weird errors and my conclusion was that it does not deal with `mut` state, it's merely for initializing constants. But let me retry, it's possible I screwed up.

Reply:
`DELTAS` does not have to be `mut`, does it? It's only `mut` here so you can initialize it. This is exactly what `lazy_static` is supposed to solve.

Reply:
It does have to mutable, because we accumulate the deltas in this vector. It works as long as there is no `mut` involved, but once I change the declaration the compiler bails out:
```
error: no rules expected the token `mut`
 --> src/main.rs:6:12
  |
6 |     static mut ref DELTAS: Mutex<Vec<usize>> = {
  |            ^^^ no rules expected this token in macro call
```

```
error: no rules expected the token `DELTAS`
 --> src/main.rs:6:20
  |
6 |     static ref mut DELTAS: Mutex<Vec<usize>> = {
  |                    ^^^^^^ no rules expected this token in macro call
```

Reply:
We clarified that in person. I'll change the code to use `lazy_static` :)